app/generic/definitions.py

```diff
@@ -0,0 +1,390 @@
+    id: str = Field(..., example="5Q1Zz9foMeAA8Q7mrmzCfZ:3:CL:7:default")
+    tag: str = Field(..., example="default")
+    schema_id: str = Field(..., example="CXQseFxV34pcb8vf32XhEa:2:test_schema:0.3")
+
+
+class CreateSchema(BaseModel):
+    name: str = Field(..., example="test_schema")
+    version: str = Field(..., example="0.3.0")
+    attribute_names: List[str] = Field(..., example=["speed"])
+
+
+class CredentialSchema(BaseModel):
+    id: str = Field(..., example="CXQseFxV34pcb8vf32XhEa:2:test_schema:0.3")
+    name: str = Field(..., example="test_schema")
+    version: str = Field(..., example="0.3.0")
+    attribute_names: List[str] = Field(..., example=["speed"])
+
+
+def _credential_schema_from_acapy(schema: ModelSchema):
+    return CredentialSchema(
+        id=schema.id,
+        name=schema.name,
+        version=schema.version,
+        attribute_names=schema.attr_names,
+    )
+
+
+def _credential_definition_from_acapy(credential_definition: AcaPyCredentialDefinition):
+    return CredentialDefinition(
+        id=credential_definition.id,
+        tag=credential_definition.tag,
+        schema_id=credential_definition.schema_id,
+    )
+
+
+@router.get("/credentials", response_model=List[CredentialDefinition])
+async def get_credential_definitions(
+    issuer_did: Optional[str] = None,
+    credential_definition_id: Optional[str] = None,
+    schema_id: Optional[str] = None,
+    schema_issuer_did: Optional[str] = None,
+    schema_name: Optional[str] = None,
+    schema_version: Optional[str] = None,
+    aries_controller: AcaPyClient = Depends(agent_selector),
+):
+    """
+    Retrieve credential definitions the current agent created.
+
+    Parameters:
+    -----------
+    issuer_did: str (Optional)\n
+    credential_definition_id: str (Optional)\n
+    schema_id: str (Optional)\n
+    schema_issuer_id: str (Optional)\n
+    schema_version: str (Optional)\n
+
+    Returns:
+    ---------
+    The created credential definitions.
+    """
+    # Get all created credential definition ids that match the filter
+    response = await aries_controller.credential_definition.get_created_cred_defs(
+        issuer_did=issuer_did,
+        cred_def_id=credential_definition_id,
+        schema_id=schema_id,
+        schema_issuer_did=schema_issuer_did,
+        schema_name=schema_name,
+        schema_version=schema_version,
+    )
+
+    # Initiate retrieving all credential definitions
+    credential_definition_ids = response.credential_definition_ids or []
+    get_credential_definition_futures = [
+        aries_controller.credential_definition.get_cred_def(
+            cred_def_id=credential_definition_id
+        )
+        for credential_definition_id in credential_definition_ids
+    ]
+
+    # Wait for completion of retrieval and transform all credential definitions
+    # into response model (if a credential definition was returned)
+    credential_definition_results = await asyncio.gather(
```

Comment:
I think you can just put await in the list comprehension above

Reply:
```python
credential_definition_results = [
        await aries_controller.credential_definition.get_cred_def(
            cred_def_id=credential_definition_id
        )
        for credential_definition_id in credential_definition_ids
    ]
```

Reply:
I think this will make the retrieving of credential definitions sequential where asyncio.gather will make it run in parallel. This can have a significant impact on time it will take to finish processing